packages/orchestrator-ui-components/src/components/WFOTree/WFOTree.tsx

```diff
@@ -1,16 +1,27 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect } from 'react';
 
 import { WFOTreeBranch } from './WFOTreeBranch';
 import { TreeBlock } from '../../types';
+import { TreeContext, TreeContextType } from '../../contexts';
 
 type WFOTreeProps = {
     data: TreeBlock[];
+    depthList: number[];
 };
 
-export const WFOTree: FC<WFOTreeProps> = ({ data }) => (
-    <div style={{ width: '500px' }}>
-        {data.map((item) => (
-            <WFOTreeBranch key={item.id} item={item} level={0} />
-        ))}
-    </div>
-);
+export const WFOTree: FC<WFOTreeProps> = ({ data, depthList }) => {
+    const { setDepths } = React.useContext(TreeContext) as TreeContextType;
+
+    useEffect(() => {
+        setDepths(depthList);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
@ricardovdheijden / @DutchBen : I am not satisfied with this eslint ignore -> changes in the TreeContext will cause a re-render; so it's needed here in the deps.

Or do I miss semething? probably the downside of the Frankenstein state management...

Reply:
You might be able to fix this situation by changing the depthList state (located in the TreeContext) in a useRef.

Reply:
For now agreed; to try to improve this in upcoming tree tickets: after that decide if we want/need to refactor the Tree context provider.